pkg/services/object/acl/v2/service.go

```diff
@@ -480,16 +480,10 @@ func (p putStreamBasicChecker) Send(request *objectV2.PutRequest) error {
 			}
 		}
 
-		var sTok *sessionSDK.Object
-
-		if tokV2 := request.GetMetaHeader().GetSessionToken(); tokV2 != nil {
-			sTok = new(sessionSDK.Object)
-
-			err = sTok.ReadFromV2(*tokV2)
-			if err != nil {
-				return fmt.Errorf("invalid session token: %w", err)
-			}
-
+		sTok, err := originalSessionToken(request.GetMetaHeader())
+		if err != nil {
+			return err
+		} else if sTok != nil {
```

Comment:
why `else if` here? just `if` should be enough

Reply:
because we may `return` (linter doesn't yell btw)

Reply:
iiuc it complicates readability, moved to a separate block